c2rust-analyze/tests/filecheck/cast.rs

```diff
@@ -16,3 +16,26 @@ pub unsafe fn null_ptr() {
     let s = 0 as *mut S;
     (*s).i = 0 as *const i32;
 }
+
+#[repr(C)]
+pub struct Foo {
+    y: *mut i32,
+}
+
+extern "C" {
+    // necessary to fix the type of Foo::y
+    fn bar(f: Foo);
+}
```

Comment:
Maybe it would be nice to have a test attribute like `#[fixed_signature]` that can be used on struct fields.

Reply:
Agreed. I think that should go into a separate PR.